lib/ohai/plugins/linux/platform.rb

```diff
@@ -60,9 +60,20 @@ def get_redhatish_version(contents)
       platform get_redhatish_platform(contents)
       platform_version contents.match(/(\d\.\d\.\d)/)[0]
     elsif File.exists?("/etc/redhat-release")
-      contents = File.read("/etc/redhat-release").chomp
-      platform get_redhatish_platform(contents)
-      platform_version get_redhatish_version(contents)
+      if File.exists?('/etc/os-release') # check if Cisco
+      # don't clobber existing os-release properties, point to a different cisco file
+        contents = {}
+        File.read('/etc/os-release').split.collect {|x| x.split('=')}.each {|x| contents[x[0]] = x[1]}
+        if File.exists?(contents['CISCO_RELEASE_INFO'])
```

Comment:
Also, we should use `exist?` instead of `exists?`, which is deprecated. I think it will likely never actually be removed, but it spits out a warning if you run `ruby -w`. I'm slowly trying to fix up warnings and enable them in the tests in our codebase as I can.